build/Dockerfile

```diff
@@ -1,46 +1,289 @@
+	&& apt-get purge --auto-remove -y apt-transport-https \
+	&& rm -rf /var/lib/apt/lists/*
+
+# Uncomment the lines below if you want to install a custom CA certificate
+# COPY build/*.crt  /usr/local/share/ca-certificates/
+# RUN update-ca-certificates
+
+
+############################################# Base image for Openshift #############################################
+FROM registry.access.redhat.com/ubi8/ubi:8.3 AS openshift-base
+
+LABEL name="NGINX Ingress Controller" \
+	description="The Ingress controller is an application that runs in a cluster and configures an HTTP load balancer according to Ingress resources." \
+	summary="The Ingress controller is an application that runs in a cluster and configures an HTTP load balancer according to Ingress resources." \
+	io.openshift.tags="nginx,ingress-controller,ingress,controller,kubernetes,openshift" \
+	maintainer="NGINX Docker Maintainers <docker-maint@nginx.com>" \
+	vendor="NGINX Inc <kubernetes@nginx.com>"
+
+RUN set -x \
+	&& groupadd --system --gid 101 nginx \
+	&& useradd --system --gid nginx --no-create-home --home-dir /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx
+
+
+RUN mkdir licenses
+COPY --chown=nginx:0 LICENSE /licenses
+
+
+############################################# Base image for Openshift OSS #############################################
+FROM openshift-base AS openshift
+ARG NGINX_VERSION
+
+RUN echo "[nginx]" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "name=nginx repo" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "baseurl=https://nginx.org/packages/mainline/rhel/8/\$basearch/" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "gpgcheck=0" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "enabled=1" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "module_hotfixes=true" >> /etc/yum.repos.d/nginx.repo \
+	&& yum install -y nginx-${NGINX_VERSION}
+
+
+############################################# Base image for Openshift with NGINX Plus #############################################
+FROM openshift-base AS openshift-plus
+
+ENV NGINX_PLUS_VERSION 23-1.el8.ngx
+ARG IC_VERSION
+
+RUN rpm --import https://nginx.org/keys/nginx_signing.key \
+	&& echo "[nginx-plus]" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "name=nginx-plus repo" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "baseurl=https://plus-pkgs.nginx.com/centos/8/\$basearch/" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientcert=/etc/ssl/nginx/nginx-repo.crt" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientkey=/etc/ssl/nginx/nginx-repo.key" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "gpgcheck=1" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "enabled=1" >> /etc/yum.repos.d/nginx-plus-8.repo
+
+# Install NGINX Plus
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	yum install -y ca-certificates nginx-plus-${NGINX_PLUS_VERSION}
```

Comment:
do we need to do clean up like here https://github.com/nginxinc/kubernetes-ingress/blob/master/build/openshift/DockerfileForPlus#L50 ?

Reply:
`/etc/ssl/nginx` is empty because we use the mount